ingestors/npm.go

```diff
@@ -1,154 +1,116 @@
 package ingestors
 
 import (
-	"context"
 	"fmt"
-	"time"
+	"io"
+	"strconv"
 
+	"github.com/buger/jsonparser"
 	log "github.com/sirupsen/logrus"
 
-	kivik "github.com/go-kivik/kivik/v4"
-
-	_ "github.com/go-kivik/kivik/v4/couchdb"
 	"github.com/librariesio/depper/data"
 )
 
-const NPMRegistryHostname = "https://replicate.npmjs.com"
-const NPMRegistryDatabase = "registry"
+const npmSchedule = "1-59 * * * *"
+const npmIndexUrl = "https://replicate.npmjs.com/registry/_changes"
 
-// Delay between attempts at initial connection to Changes() feed
-const ConnectRetryDelaySeconds = 60 * 5
+// Arbitrary sequence number from 5/20/2025, just to make sure we
+// don't fallback to "0". Via https://replicate.npmjs.com/registry/.
+const fallbackSequence = 42540679
+const perPage = 10000
 
-// Delay between attemps to fix broken connection to Changes() feed
-const ReconnectRetryDelaySeconds = 5
+// Limit number of pages per run in case we hit a bad sequence number / backfill.
+const maxResultsLength = 100000
```

Comment:
This seems very high

Reply:
Updated it to 1 page of 10k for every 5 minute run, which gives us headroom for almost 3 million events per day.